### PR TITLE
Fix performance bug when pasting large amounts of text

### PR DESCRIPTION
**Pull request by jasonleenaylor**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/92)

### CODE REVIEW

File: src/SIL.LCModel.Core/Text/TsStrBldr.cs [hunk 1]
```diff
@@ -319,5 +319,59 @@ public void Clear()
...
+			}
+
+			public void Clear()
+			{
+				_cachedString = null;
+				_stringBuilder.Clear();
+			}
+
+			public void Remove(int ichMin, int i)
+			{
```
Comment (ermshiperete) [inline src/SIL.LCModel.Core/Text/TsStrBldr.cs]: You could add an assert like `Debug.Assert(_cachedString == null || _stringBuilder.Length == 0);`

File: src/SIL.LCModel.Core/Text/TsStrBldr.cs [hunk 2]
```diff
@@ -319,5 +319,59 @@ public void Clear()
...
+				{
+					_stringBuilder.Append(_cachedString);
+					_cachedString = null;
+				}
+
+				_stringBuilder.Remove(ichMin, i);
+			}
+
+			public void Insert(int ichMin, string insertText)
+			{
```
Comment (ermshiperete) [inline src/SIL.LCModel.Core/Text/TsStrBldr.cs]: You could add an assert like `Debug.Assert(_cachedString == null || _stringBuilder.Length == 0);`

**Comment by jasonleenaylor** [comment]
Yes, when we are pasting a large string into the baseline it ends up calling .ToString() on the builder multiple times for every character which is in the pasted text. For 'small' strings this is fine, but for large strings it is expensive to make the copies.

**Comment by jasonleenaylor** [comment]
LGTM based on Eberhard's coment

**Pull request merged**